bot/exts/moderation/voice_gate.py

```diff
@@ -0,0 +1,154 @@
+import logging
+from contextlib import suppress
+from datetime import datetime, timedelta
+
+import discord
+from dateutil import parser
+from discord import Colour
+from discord.ext.commands import Cog, Context, command
+
+from bot.api import ResponseCodeError
+from bot.bot import Bot
+from bot.constants import Channels, Event, MODERATION_ROLES, Roles, VoiceGate as GateConf
+from bot.decorators import has_no_roles, in_whitelist
+from bot.exts.moderation.modlog import ModLog
+from bot.utils.checks import InWhitelistCheckFailure
+
+log = logging.getLogger(__name__)
+
+FAILED_MESSAGE = (
+    """You are not currently eligible to use voice inside Python Discord for the following reasons:\n\n{reasons}"""
+)
+
+MESSAGE_FIELD_MAP = {
+    "verified_at": f"have been verified for less {GateConf.minimum_days_verified} days",
+    "voice_banned": "have an active voice ban infraction",
+    "total_messages": f"have sent less than {GateConf.minimum_messages} messages",
+}
+
+
+class VoiceGate(Cog):
+    """Voice channels verification management."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @property
+    def mod_log(self) -> ModLog:
+        """Get the currently loaded ModLog cog instance."""
+        return self.bot.get_cog("ModLog")
+
+    @command(aliases=('voiceverify', 'vverify', 'voicev', 'vv'))
+    @has_no_roles(Roles.voice_verified)
+    @in_whitelist(channels=(Channels.voice_gate,), redirect=None)
+    async def voice_verify(self, ctx: Context, *_) -> None:
```

Comment:
We don't need this many aliases.
```suggestion
    @command(aliases=('voiceverify',))
    @has_no_roles(Roles.voice_verified)
    @in_whitelist(channels=(Channels.voice_gate,), redirect=None)
    async def voice_verify(self, ctx: Context, *_) -> None:
```